pkg/api/graph/graph.go

```diff
@@ -404,6 +447,10 @@ func (g Graph) SubgraphWithNodes(nodes []graph.Node, fn EdgeFunc) Graph {
 	for _, node := range nodes {
 		out.internal.AddNode(node)
 	}
+	fmt.Println("Edges under review:")
+	for _, edge := range g.internal.Edges() {
```

Comment:
Logged down edges sizes for all the subgraphs I create during the process:
```
[vagrant@openshiftdev sample-app]$ oc get bc
NAME                           TYPE      SOURCE
ruby-hello-world               Docker    https://github.com/openshift/ruby-hello-world
ruby-sample-build-invalidtag   Source    git://github.com/openshift/ruby-hello-world.git
ruby-sample-build-validtag     Source    git://github.com/openshift/ruby-hello-world.git
```

Initial graph: 9 edges
Sub#1 - clipped out inbounds to ist of interest: 6 edges
Sub#2 - strong edition of sub#1: **6 edges** (should be 12)
Sub#3 - final partitioned dep graph (varies depending on what comes out of sub#2)

It looks like [g.Edges()](https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/gonum/graph/concrete/directed.go#L252) doesn't account for ReferencedByEdgeKind edges in the final edge list size.